File: AutomatonAlgorithm/src/automata.java
//import java.util.ArrayList;
//
///**
// * Created by jenamorado on 7/25/16.
// */
//public class automata {
//
//    public static void main(String[] args) {
//        Automaton a = new Automaton();
//        States state1 = new States();
//        States states2 =new States();
//        state1.isInitialState = true;
//        state1.name="Initial State";
//        state1.isAcceptanceState = false;
//        states2.isInitialState = false;
//        states2.name="Acceptance State";
//        state1.isInitialState = false;
//        states2.isAcceptanceState = false;
//        Transition transition1 = new Transition();
//        transition1.symbol='1';
//        transition1.origin = state1.name;
//        transition1.destiny = states2.name;
//        state1.transitions = new ArrayList<>();
//        state1.transitions.add(transition1);
//        Transition transition2 = new Transition();
//        transition2.symbol='0';
//        transition2.origin = states2.name;
//        transition2.destiny = state1.name;
//        states2.transitions = new ArrayList<>();
//        states2.transitions.add(transition2);
//        a.states = new ArrayList<>();
//        a.states.add(state1);
//        a.states.add(states2);
//        System.out.println(a.evaluate("1"));
//    }
//}

File: AutomatonAlgorithm/src/KotlinAlgorithm/automata.kt
/**
 * Created by Jobarah on 7/25/2016.
 */
fun main(arr : Array<String>){
    var a =  Automaton()
    val state1 = States("q0",true,false)
    val states2 = States("q1",false,true)
    var transition1 = Transition('0',"q0","q1")
    var transition2 = Transition('1',"q1","q0")
    state1._transition.add(transition1)
    states2._transition.add(transition2)
    a.states.add(state1)
    a.states.add(states2)
    a.alphabet.add('0')
    a.alphabet.add('1')
    println(a.evaluate("0"))
}
